### PR TITLE
Link: updated Docs to use autogenerated prop table

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Link: updated Docs to use autogenerated prop table

#### Why?

Gestalt is moving in this direction, all components must autogenerate prop tables

### CODE REVIEW

File: docs/pages/avatar.js [hunk 1]
```diff
@@ -43,7 +43,6 @@ export default function AvatarPage({ generatedDocGen }: {| generatedDocGen: DocG
       />
 
       <GeneratedPropTable generatedDocGen={generatedDocGen} />
-
```
Comment (dangerismycat) [inline docs/pages/avatar.js:46]: this seems like an unrelated change

File: docs/pages/link.js [hunk 2]
```diff
@@ -183,14 +66,15 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
...
-        description={`
-    Use accessibilitySelected and role when using it as a Tab.
-  `}
-        name="Accessible Tab Link"
-        defaultCode={`
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Accessible Tab Link"
+          description="Use accessibilitySelected and role when using it as a Tab."
```
Comment (dangerismycat) [inline docs/pages/link.js:73]: Can you add backticks around the prop names? Also "it" isn't clear, and "Tab" doesn't need to be capitalized since we're not referring to our component:
"Use `accessibilitySelected` and `role="tab"` when using Link as a tab."

File: docs/pages/link.js [hunk 3]
```diff
@@ -222,11 +106,40 @@ function TabExample() {
...
-        id="Permutations"
-        name="Permutations: tapStyle and hoverStyle"
-        defaultCode={`
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Link and Text"
+          description="You should wrap Link inside of a [Text](https://gestalt.pinterest.systems/text) component to get the correct font & underline color."
```
Comment (dangerismycat) [inline docs/pages/link.js:115]: This is a little awkward. Perhaps
"Use Link within Text to get the correct font and underline color"

File: docs/pages/link.js [hunk 4]
```diff
@@ -222,11 +106,40 @@ function TabExample() {
...
+    <Link href="https://pinterest.com">
+      <Box padding={2}>Correct underline color: Pinterest.com</Box>
+    </Link>
+  </Text>
+</Box>`}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="tapStyle and hoverStyle"
+          description="Use accessibilitySelected and role when using it as a Tab."
```
Comment (dangerismycat) [inline docs/pages/link.js:138]: ditto here

File: packages/gestalt/src/Link.js [hunk 5]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
 
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like `Click Here,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See the [Accessibility](https://gestalt.pinterest.systems/link#accessibility) guidelines for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:24]: The link isn't correct: "Accessibility" needs to be capitalized to hit the correct anchor. (Also a nit: I would think to link "Accessibility guidelines", not just the first word)

File: packages/gestalt/src/Link.js [hunk 6]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   accessibilityLabel?: string,
+  /**
+   * Use accessibilitySelected and role when using it as a Tab. See the [Accessibility](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) guidelines for more information.
+   */
   accessibilitySelected?: boolean,
+  /**
+   * Link is a wrapper around components (or children), most commonly [Text](https://gestalt.pinterest.systems/text), so that they become hiperlinks. See the [Link and Text](https://gestalt.pinterest.systems/link#Link-and-Text) variant for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:32]: typo: "hyperlinks"

This wording is also a bit confusing, given that we recommend that Text wraps around Link, not Link wrapping around Text. Perhaps it would be more clear if we refer to "text" — the generic, not our component.

File: packages/gestalt/src/Link.js [hunk 7]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   children?: Node,
+  /**
+   * When `undeline` is supplied, Link's text is underlined on hover. See the [tapStyle and hoverStyle](https://gestalt.pinterest.systems/link#tapStyle-and-hoverStyle) variant for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:36]: typo: "underline"

File: packages/gestalt/src/Link.js [hunk 8]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
 
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like `Click Here,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See the [Accessibility](https://gestalt.pinterest.systems/link#accessibility) guidelines for more information.
+   */
   accessibilityLabel?: string,
+  /**
+   * Use accessibilitySelected and role when using it as a Tab. See the [Accessibility](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) guidelines for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:28]: The props should have backticks, and Tab shouldn't be capitalized

File: packages/gestalt/src/Link.js [hunk 9]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+  /**
+   * When `undeline` is supplied, Link's text is underlined on hover. See the [tapStyle and hoverStyle](https://gestalt.pinterest.systems/link#tapStyle-and-hoverStyle) variant for more information.
+   */
   hoverStyle?: 'none' | 'underline',
+  /**
+   * The URL that the hyperlink points to.
+   */
   href: string,
+  /**
+   * Id attribute of the anchor tag
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:44]: This should mention something about the id being unique

File: packages/gestalt/src/Link.js [hunk 10]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   id?: string,
+  /**
+   * Properly positions Link relative to an inline element, such as [Text](https://gestalt.pinterest.systems/text), using the inline property.
+   */
   inline?: boolean,
+  /**
+   * Callback triggered when the user blurs Link.
+   */
   onBlur?: AbstractEventHandler<SyntheticFocusEvent<HTMLAnchorElement>>,
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:54]: The AbstractEventHandlers need to be inlined

File: packages/gestalt/src/Link.js [hunk 11]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+  /**
+   * Id attribute of the anchor tag
+   */
   id?: string,
+  /**
+   * Properly positions Link relative to an inline element, such as [Text](https://gestalt.pinterest.systems/text), using the inline property.
+   */
   inline?: boolean,
+  /**
+   * Callback triggered when the user blurs Link.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:52]: Perhaps "…when the element loses focus" — the user isn't always the one shifting focus

File: packages/gestalt/src/Link.js [hunk 12]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
   onBlur?: AbstractEventHandler<SyntheticFocusEvent<HTMLAnchorElement>>,
+  /**
+   * Callback fired when Link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/onlinknavigationprovider) to learn more about link navigation.
+   */
   onClick?: AbstractEventHandler<
     SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
     {| dangerouslyDisableOnNavigation: () => void |},
   >,
+  /**
+   * Callback triggered when the user focuses Link.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:63]: Perhaps "…when the element gains focus"

File: packages/gestalt/src/Link.js [hunk 13]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   onFocus?: AbstractEventHandler<SyntheticFocusEvent<HTMLAnchorElement>>,
+  /**
+   * Forward the ref to the underlying anchor element
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:67]: This wording doesn't quite make sense. Perhaps something like
"Ref that is forwarded to the underlying anchor element."

File: packages/gestalt/src/Link.js [hunk 14]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
+  ref?: Ref<'a'>,
+  /**
+   * Establishes the relationship of the linked URL. Use rel=”nofollow” for offsite links to inform search engines to ignore and not follow them
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:71]: backticks around the `rel` example

File: packages/gestalt/src/Link.js [hunk 15]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   rel?: 'none' | 'nofollow',
+  /**
+   * When supplied, Link acts a Tab. See the [Accessible Tab Link](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) for more information.
+   */
   role?: 'tab',
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:77]: This is coming up as "literal" for some reason
Screen Shot 2022-02-04 at 1 50 11 PM

File: packages/gestalt/src/Link.js [hunk 16]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+  /**
+   * Forward the ref to the underlying anchor element
+   */
+  ref?: Ref<'a'>,
+  /**
+   * Establishes the relationship of the linked URL. Use rel=”nofollow” for offsite links to inform search engines to ignore and not follow them
+   */
   rel?: 'none' | 'nofollow',
+  /**
+   * When supplied, Link acts a Tab. See the [Accessible Tab Link](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:75]: "Tab" shouldn't be capitalized

File: packages/gestalt/src/Link.js [hunk 17]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   role?: 'tab',
+  /**
+   * Sets a border radius for Link. Select a rounding option that aligns with its children.
+
+Options are "circle" or "pill" for fully rounded corners or 0-8 representing the radius in boints.
+   */
   rounding?: Rounding,
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:83]: This needs to be inlined

File: packages/gestalt/src/Link.js [hunk 18]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   rel?: 'none' | 'nofollow',
+  /**
+   * When supplied, Link acts a Tab. See the [Accessible Tab Link](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) for more information.
+   */
   role?: 'tab',
+  /**
+   * Sets a border radius for Link. Select a rounding option that aligns with its children.
+
+Options are "circle" or "pill" for fully rounded corners or 0-8 representing the radius in boints.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:81]: This sentence isn't necessary, it's just repeating the type displayed above it

File: packages/gestalt/src/Link.js [hunk 19]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+   */
   rounding?: Rounding,
+  /**
+   * When `compress` is supplied, Link is compressed on click. See the [tapStyle and hoverStyle](https://gestalt.pinterest.systems/link#tapStyle-and-hoverStyle) variant for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:85]: Perhaps "visually compressed" is more clear?

File: packages/gestalt/src/Link.js [hunk 20]
```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
...
+
+'null' opens the anchor in the same window.
+'blank' opens the anchor in a new window.
+'self' opens an anchor in the same frame.
+   */
   target?: null | 'self' | 'blank',
-  // private props to be internally used, therefore, not documented
-  disabled?: boolean,
+  /**
+   * Private prop
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:97]: the comment won't be displayed, but we can document for ourselves what this prop is doing

File: docs/pages/avatar.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/avatar.js:46]: DONE

File: docs/pages/link.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/link.js:73]: DONE

File: docs/pages/link.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/pages/link.js:115]: DONE

File: packages/gestalt/src/Link.js (hunk 16 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:75]: DONE

File: packages/gestalt/src/Link.js (hunk 17 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:83]: DONE

File: packages/gestalt/src/Link.js (hunk 18 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:81]: DONE

File: packages/gestalt/src/Link.js (hunk 19 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:85]: DONE

File: packages/gestalt/src/Link.js (hunk 6 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:32]: DONE

File: packages/gestalt/src/Link.js (hunk 7 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:36]: DONE

File: packages/gestalt/src/Link.js (hunk 9 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:44]: DONE

File: packages/gestalt/src/Link.js (hunk 11 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:52]: DONE

File: packages/gestalt/src/Link.js (hunk 13 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:67]: DONE

File: packages/gestalt/src/Link.js (hunk 17 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:81]: DONE

File: packages/gestalt/src/Link.js (hunk 20 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js]: Extended but disabled should be deprecated, it's an old prop from when InternalLink wasn't being used.

File: packages/gestalt/src/Link.js (hunk 15 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:78]: FIXED

File: docs/components/GeneratedPropTable.js [hunk 21]
```diff
@@ -101,7 +101,12 @@ export default function GeneratedPropTable({
       );
 
       // Trim leading `|`
-      const transformedType = (flowType.raw?.replace(/^\|/, '').trim() ?? flowType.name ?? '')
+      const transformedType = (
+        flowType.raw?.replace(/^\|/, '').trim() ??
+        flowType?.value ??
```
Comment (AlbertCarreras) [inline docs/components/GeneratedPropTable.js:106]: we need value so that single string values don't show up as 'literal'

Comment (dangerismycat) [reply docs/components/GeneratedPropTable.js:106]: good find!

File: docs/components/GeneratedPropTable.js [hunk 22]
```diff
@@ -112,7 +117,6 @@ export default function GeneratedPropTable({
         // Replace "Ref" with "React.Ref" to match docs convention
         // Includes `<` to avoid picking up `HTMLDivElement` and similar
         .replace(/Ref</g, 'React.Ref<');
-
```
Comment (dangerismycat) [inline docs/components/GeneratedPropTable.js:115]: nit: leaving whitespace like this is nice, makes the code easier to read

File: docs/pages/link.js [hunk 23]
```diff
@@ -183,14 +66,15 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
...
-        description={`
-    Use accessibilitySelected and role when using it as a Tab.
-  `}
-        name="Accessible Tab Link"
-        defaultCode={`
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Accessible Tab Link"
+          description="Use `accessibilitySelected` and `role='Tab'` when using Link as a tab."
```
Comment (dangerismycat) [inline docs/pages/link.js:73]: typo: `role="tab"`

File: docs/pages/link.js [hunk 24]
```diff
@@ -222,11 +106,40 @@ function TabExample() {
...
-        id="Permutations"
-        name="Permutations: tapStyle and hoverStyle"
-        defaultCode={`
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Link and Text"
+          description="Use Link within [text](https://gestalt.pinterest.systems/text) to get the correct font and underline color."
```
Comment (dangerismycat) [inline docs/pages/link.js:115]: If we're referring to our Text component, it should be capitalized

File: docs/pages/link.js [hunk 25]
```diff
@@ -222,11 +106,40 @@ function TabExample() {
...
+    <Link href="https://pinterest.com">
+      <Box padding={2}>Correct underline color: Pinterest.com</Box>
+    </Link>
+  </Text>
+</Box>`}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="tapStyle and hoverStyle"
+          description="Use `accessibilitySelected` and `role='Tab'` when using Link as a tab."
```
Comment (dangerismycat) [inline docs/pages/link.js:138]: same typo here

File: packages/gestalt/src/Link.js [hunk 26]
```diff
@@ -1,43 +1,104 @@
...
 
+type Rounding = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 'circle' | 'pill';
+
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like `Click Here,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See the [Accessibility guidelines](https://gestalt.pinterest.systems/link#Accessibility) for more information.
+   */
   accessibilityLabel?: string,
+  /**
+   * Use `accessibilitySelected` and role when using it as a tab. See the [Accessibility guidelines](https://gestalt.pinterest.systems/link#Accessible-Tab-Link) for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:29]: `role` also needs backticks

File: packages/gestalt/src/Link.js [hunk 27]
```diff
@@ -1,43 +1,104 @@
...
+import getRoundingClassName from './getRoundingClassName.js';
 import focusStyles from './Focus.css';
 import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
 
+type Rounding = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 'circle' | 'pill';
+
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like `Click Here,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See the [Accessibility guidelines](https://gestalt.pinterest.systems/link#Accessibility) for more information.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:25]: The formatting is a little confusing here. We typically use backticks for code-related things, so the example text ("Click here", "Read more") should use quotes instead, and `aria-label` should have backticks

File: packages/gestalt/src/Link.js [hunk 28]
```diff
@@ -1,43 +1,104 @@
...
+  /**
+   * Callback triggered when the element gains focus.
+   */
+  onFocus?: ({| event: SyntheticFocusEvent<HTMLAnchorElement> |}) => void,
+  /**
+   * Ref that is forwarded to the underlying anchor element.
+   */
+  ref?: Ref<'a'>,
+  /**
+   * Establishes the relationship of the linked URL. Use `rel=”nofollow”` for offsite links to inform search engines to ignore and not follow them.
```
Comment (dangerismycat) [inline packages/gestalt/src/Link.js:72]: It looks like the quotes got replaced by "smart quotes", which can be problematic (especially if anyone copy/pastes into code)

File: packages/gestalt/src/Link.js (hunk 20 above)
Comment (dangerismycat) [reply packages/gestalt/src/Link.js]: Ah cool, hence not renaming to `_disabled`. I assume you'll take care of that deprecation in a future PR?

**Review by dangerismycat — APPROVED** [review]

File: docs/components/GeneratedPropTable.js (hunk 22 above)
Comment (AlbertCarreras) [reply docs/components/GeneratedPropTable.js:115]: DONE

File: docs/pages/link.js (hunk 23 above)
Comment (AlbertCarreras) [reply docs/pages/link.js:73]: DONE

File: docs/pages/link.js (hunk 24 above)
Comment (AlbertCarreras) [reply docs/pages/link.js:115]: DONE

File: docs/pages/link.js (hunk 25 above)
Comment (AlbertCarreras) [reply docs/pages/link.js:138]: DONE

File: packages/gestalt/src/Link.js (hunk 27 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js:25]: DONE

File: packages/gestalt/src/Link.js (hunk 28 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Link.js]: DONE
